test/remote/device_tests.js

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var test = require('../ptaptest')
+var TestServer = require('../test_server')
+var Client = require('../client')
+var config = require('../../config').getProperties()
+var crypto = require('crypto')
+
+TestServer.start(config)
+.then(function main(server) {
+  test(
+    'device registration',
+    function (t) {
+      var email = server.uniqueEmail()
+      var password = 'test password'
+      var deviceInfo = {
+        name: 'test device',
+        type: 'desktop',
+        pushCallback: 'https://foo/bar',
+        pushPublicKey: crypto.randomBytes(32).toString('hex')
+      }
+      var client
+      return Client.create(config.publicUrl, email, password, {
+        device: deviceInfo
+      })
+      .then(
+        function (result) {
+          client = result
+          t.ok(client.authAt, 'authAt was set')
+          t.ok(client.device.id, 'device.id was set')
+          t.ok(client.device.createdAt > 0, 'device.createdAt was set')
+          t.equal(client.device.name, deviceInfo.name, 'device.name is correct')
+          t.equal(client.device.type, deviceInfo.type, 'device.type is correct')
+          t.equal(client.device.pushCallback, deviceInfo.pushCallback, 'device.pushCallback is correct')
+          t.deepEqual(client.device.pushPublicKey, deviceInfo.pushPublicKey, 'device.pushPublicKey is correct')
+        }
+      )
+      .then(
+        function () {
+          return client.devices()
```

Comment:
Let's also add a test for creating the device as a separate step after login, rather than inline in the account-creation call.  IIUC this is how you're planning to do the initial browser implementation so we should test it explicitly.